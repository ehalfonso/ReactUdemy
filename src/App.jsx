import { CounterApp } from "./components/CounterApp";

export const App = () => {
  return (
    <>
      <CounterApp value={123} />
    </>
  );
};
