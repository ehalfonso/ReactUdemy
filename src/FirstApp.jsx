import { PropTypes } from "prop-types";

export const FirstApp = ({ title, subTitle, name: name }) => {
  return (
    <>
      <div>
        <h1 data-testid="test=title">{title}</h1>
        <h2>{subTitle}</h2>
        <h2>{subTitle}</h2>
        <p>{name}</p>
      </div>
    </>
  );
};

FirstApp.protoType = {
  title: PropTypes.string.required,
  subTitle: PropTypes.string.required,
};

FirstApp.defaultProps = {
  name: "Eric Herrera",
  subTitle: "No hay subtitulo",
  //   title:"No hay titulo"
};
