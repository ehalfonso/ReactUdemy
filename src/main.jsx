import React from "react";
import ReactDOM from "react-dom/client";
import { App } from "./App.jsx";
import "./style/App.css";
import { FirstApp } from "./FirstApp.jsx";
import { CounterApp } from "./components/CounterApp.jsx";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp title="Hello, I'm Iron Man" /> */}
    <CounterApp value={20} />
  </React.StrictMode>
);
