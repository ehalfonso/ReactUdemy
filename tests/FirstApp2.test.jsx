/**@jest-ebvironment jsdom*/
import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Test in <FirstApp/>", () => {
  const title = "Hello,I'm Iron man";
  const subtitle = "I'm subtitue";
  test("should match with snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });
  test('should show a message "Hello, I\'m Iron MAn"', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
  test("should show the title in a h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
  test("should show the sent subtitle by props", () => {
    render(<FirstApp title={title} subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
