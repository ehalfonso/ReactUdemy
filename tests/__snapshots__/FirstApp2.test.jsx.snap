// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp/> should match with snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test=title"
    >
      Hello,I'm Iron man
    </h1>
    <h2>
      No hay subtitulo
    </h2>
    <h2>
      No hay subtitulo
    </h2>
    <p>
      Eric Herrera
    </p>
  </div>
</div>
`;
